### ISSUE TITLE
Update base Dolphin version?

### ISSUE THREAD

**Issue by dantman** [CarlKenner/dolphin#123]
Dolphin seems to have been improved significantly since the last build of Dolphin VR. It would be nice to see a new build based on the more recent version of Dolphin.

Of a particular unfortunate note. I wanted to try playing the LoZ: WindWaker randomizer in VR. Unfortunately the randomizer version gets a black screen when it is loaded in old versions of Dolphin, you need one of the recent versions to work. And unfortunately Dolphin VR is based on a version of Dolphin that is too old to run LoZ: WindWaker randomizer.

**Comment by dantman** [comment]
Yes. I also saw his comments lamenting that his programming skills in this area aren't the best. And it's not ideal to put all of VR emulation on his shoulders. The most ideal thing would be if some knowledgeable Dolphin emulator contributors could review the code he's made and merge it back into Dolphin itself.

**Comment by dreimer1986** [comment]
Old report, I know... But I hate duplicates. Just a small explaination for the rest in here.
According to many sources like the Dolphin forums and reddit, too the main reason for not developing Dolphin VR any further was a licensing problem Carl did not see/care about at first.
All libraries for Oculus and Vive support were incompatible to the GPL v2+ used as license for Dolphin. This is the reason many projects waited ages for OpenXR to finally show up (FOSS standardized lib for VR !and! GPL compatible!). The devs of Dolphin had nothing better to do than lamenting about some 3rd party libs used in Dolphin VR even when all of Carl's work was still open source as the license forces you to. All the great work already done and the astonishing look of games like Zelda WW and F-Zero FX created by it... noone cared to find a solution.
It was more like in these pesky *nix communities where ppl bash you for keeping your code locally as long as you still work on a update before you show it off @github as if you would break your own project's licensing by doing so! We germans call that Kindergarten and english ppl know the word, too! Well to conclude the story, Carl seems to be a bit like me and reacted with aggressive sarcasm combined with some politics discussion causing a ban in the Dolphin forums and a few other less useful places. This was the end.

Maybe Carl wants to resurrect it based on the OpenXR runtime which is out by now, but I cannot answer you that. I just can tell you that no other dev shows much interest right now in the Dolphin bubble.

And yes, he seems to be alive according to his other projects in here which still get updates quite often and even in recent times.

Hey Carl, if you read this, pleeease answer us. As you see there are still maaany ppl on your side and many still use the old Dolphin VR build today to play a nice session of F-Zero on VR. I checked the Culling Code reddit these days, even there is still updates for games to look better... Come back plz

**Comment by isamu99** [comment]
> 
> 
> Old report, I know... But I hate duplicates. Just a small explaination for the rest in here.

<snip>

Mr dreimer1986.....I want to give you a very humble and sincere thank you for offering this explanation regarding the Dolphin VR situation. I too, am a big fan of the old Dolphin VR fork that Carl worked on continue to use it occasionally to play F-ZeroGX. 

I agree with you 100%. Carl.....we really would like you to come back and resurrect this project. We are big fans of your work. However, as good as F-Zero GX looks, there are quite a few bugs with camera angles and such, that still remain. Carl, if you happen to read this, please let us know you're alive brother. Tell us how you feel about this project and whether or not you intend to come back to it some day. Would really appreciate it.

**Comment by CarlKenner** [comment]
I can get around the licensing problem by just writing my own headers that call the Oculus DLLs directly. The Oculus DLLs are obviously system libraries according to GPL, just like DirectX, XInput, or even just System32. I just need to get around to it. I'm banned from contributing or commenting on the main Dolphin repository, but thanks to open source, that doesn't affect my ability to work on my own fork or pull their changes.

My VR development computer stopped working, but I removed the hard disk and it seems to be working again. Also, I've been distracted and busy with other things. And I've been putting off working on Dolphin VR because the latest versions handle frame timing, frame buffers, and things like that differently, and I need to rethink my approach to how I was rendering it in VR. But I will stop procrastinating and get back to work on it.

But first, I want to finish some work on Doom 3 BFG VR: Fully Possessed. I want to get dual-wielding, bonus characters, cutscenes, and hand interactions working properly, and fix some bugs.

**Comment by isamu99** [comment]
Whoah!!!! He LIVES!!!!!!!!!!! Wow, color me surprised! Thanks so much for your reply Carl. Good to know your still keeping busy with programming and other things. Yes, we really appreciate your work, and we will patiently wait until you return to working on Dolphin. Do what you have to do man. We'll be here. 

I personally don't plan on getting back into playing any VR games until later this year when the Pimax 8K-X is released. For those of you who don't know, it will be the world's first VR headset with two TRUE native 4K displays and a wide 270º FOV. I can't imagine what the experience will be like playing F-ZeroGX on such a headset. *drools*....

https://www.youtube.com/watch?v=NdLtsjX0dPI&t

**Comment by dreimer1986** [comment]
Hey @CarlKenner, great to hear from you :D And even more great that you are even thinking about resuming your work afrer some more work on Doom 3 BFG VR (nice work too). If you need some tester with some other VR headset... I bought a Valve Index and patiently (as much as possible) wait for it to arrive end of the month.

**Comment by GeekyGami** [comment]
Hey @CarlKenner , great to hear from you!
I had been waiting with bated breaths, as I saw that OpenXR had started being shipped to headset developers.

Are you planning on OpenXR implementation once OpenXR finally hits actual devs?
This would already let you work around the GPLV2 license issues, as far as I understand. It'd also mean greater overall compatibility, and a possibility of merging Dolphin VR into Dolphin itself, which could potentially mean more support from the Dolphin community for stereoscopic 3D issues and culling shenanigans.

**Comment by isamu99** [comment]
Great idea and suggestion Greeky 👍

**Comment by dreimer1986** [comment]
Well, the GPL problem was already addressed in the first line of his post. ^^ "I can get around the licensing problem by just writing my own headers that call the Oculus DLLs directly." But OpenXR is a way to go too of course.

**Comment by GeekyGami** [comment]
Right, however, previous attempts at bypassing the GPL haven't gone very well.
With this in mind, OpenXR entirely removes any doubt that this wouldn't violate the GPL, and thus is a much safer alternative long-term.

**Comment by dreimer1986** [comment]
Well... or not: https://github.com/KhronosGroup/OpenXR-Docs/issues/3

**Comment by GeekyGami** [comment]
Oh my

**Comment by GeekyGami** [comment]
https://www.khronos.org/news/press/khronos-releases-openxr-1.0-specification-establishing-a-foundation-for-the-ar-and-vr-ecosystem

Well, regardless, it's out now.
Not sure if they changed anything

**Comment by sheepdestroyer** [comment]
GPL headers for openXR not yet in 1.0 but incoming :
https://github.com/KhronosGroup/OpenXR-Docs/issues/3#issuecomment-516500835

**Comment by BiatuAutMiahn** [comment]
Any updates on this? tried doing a rebase and also just merging with dol master...lol that was a nightmare

**Comment by GeekyGami** [comment]
Hmmm. Well then, I guess we wait for Khronos. How ironic.

**Comment by dreimer1986** [comment]
So as it seems the idea has spread a bit. dolphin-emu/dolphin/pull/8380 tries to use OpenXR for VR support. With caveats and no use at all for Vive and Index users, but... he tries to. Still faaar away from what was already made possible by this fork in the past though...

**Comment by GeekyGami** [comment]
I can't wait to see where this goes. It'd be nice to have hide codes implemented into that version.

**Comment by dreimer1986** [comment]
Small update. SteamVR natively supports OpenXR now. So if OpenXR is a consideration as the other PR tries/tried to do, Vive and Index will be working then from start. The other PR is a proof of concept right now and far away from what was achieved in here already in the past years ago.

**Comment by CarlKenner** [comment]
I have been working on redoing VR support for the latest versions of Dolphin. The architecture has changed significantly, so I'm going to redo some things from scratch. I have some cool new features mostly working, like playing games on the screen of a virtual cinema.
OpenXR is just a third API that I need to support, along with OpenVR and Oculus. There's nothing special about it and no reason for Dolphin devs to insist on it besides some weird dogma. I haven't been able to develop for OpenXR yet because there weren't any drivers for it for the Rift or HTC Vive. If it is working now via SteamVR, then I'll give it a go.
There are some issues with using OpenVR in an emulator that has framebuffers that the emulator can render to whenever it wants and then present whenever it wants, because OpenVR does not provide a way to specify what rotation matrix a framebuffer was drawn with or when the head-tracking was sampled for that frame. Even if I save the information with the framebuffer, I can't pass it on to OpenVR and OpenVR will always apply incorrect timewarp. So it would be nice if that problem is solved in OpenXR.
Hiding objects isn't really the right approach, so I've been working on better analyzing what the game is trying to do and converting that to VR. For example, I need to detect when a game is trying to do a screenspace effect, then apply that to the entire view instead of treating it like a HUD. And I need to detect when the game is trying to render shadows, and render them appropriately. And I need to detect when the game is trying to render reflections, etc. I've been making some progress. To help me do that, I have greatly improved the FIFO Analyzer: https://github.com/JosJuice/dolphin/pull/1
I've also been working on how the emulator renders to multiple views so that I can do a better job of stopping my changes to eye-view-rendering from interfering with other types of rendering. I'm thinking I might need to have 3 layers rendered at once instead of 2, and rendering to different viewports within each layer, so that there's always an EFB layer that corresponds to what the game is expecting in addition to a layer for each VR eye. I've done some experiments that suggest it will work. That will also allow better local-multiplayer support, with one player playing on the screen while another is in VR.
And I've been making the renderer multithreaded so I can render to VR at a different framerate than the game is running at, etc.
It will be a while before the new version is working well enough to upload.

**Comment by isamu99** [comment]
Hi Carl. I am pleasantly surprised to hear from you and telling us you're still working on DolphinVR. It's very encouraging to hear that you're re-doing the VR from scratch and working to improve it. Very very exciting!

There is one thing in particular I'm hoping to see in the next release, and that is support for VR headsets with a wide field of view. Headsets such as the Pimax 8K series, and now there is a new one out called the StarVR:  

https://www.youtube.com/watch?v=k5XoZxCD7qc

https://www.youtube.com/watch?v=EDGyDZvrZWA

https://www.youtube.com/watch?v=nwqFvusWPII

https://www.youtube.com/watch?v=1v23Pe51uvc

https://www.youtube.com/watch?v=HrdQfGhHt-k


Carl, when you get a chance and at your convenience, can you please watch those videos and tell me, based on your impressions, whether the StarVr Headset will be compatible with DolphinVR? This headset has a large FOV of 210º!!! Would games running in the upcoming future release of DolphinVR take full advantage of the wide FOV the StarVR One has to offer?

**Comment by dreimer1986** [comment]
Hey Carl, great to read all the news from you. One lil sidenote regarding Rift and OpenXR. https://uploadvr.com/oculus-rift-basic-openxr-support/ There is prelimary support in the Public Test branch of the Oculus Software for a while now. I have no clue how well this works though...

**Comment by sheepdestroyer** [comment]
Isn't the fact that OpenXR's license is compatible with Dolphin's the point?

**Comment by dreimer1986** [comment]
That was already answered by him: https://github.com/CarlKenner/dolphin/issues/123#issuecomment-498516103

**Comment by Lucaspec72** [comment]
it's great to hear you are working on a new version Carl, i'm looking forward to it =D

**Comment by Toblerone1** [comment]
Yes thank you so much for revisiting this man, I and so many others extremely appreciate it. I can't think of anything I've enjoyed more from VR than being able to immerse myself in older games!

**Comment by GeekyGami** [comment]
Any news of late? Haven't heard anything since Septembre, it's a new year
(Happy new year fyi)

**Comment by thatsprettygood** [comment]
His last commit on GitHub was in October 2020.. Is this normal for him?

**Comment by Lucaspec72** [comment]
he is remaking the whole thing from scratch, that probably takes a lot of work to make. i doubt he's working on it 24/7, he probably has other projects/work/school. Also keep in mind he isn't paid a penny for doing this as far as i know, so il looks more like a hobby project than anything else. (he also could of just lost interest in dolphinVR, but that sounds rather unlikely looking at the other stuff he made that looked tedious AF to do, like the doom 3 VR mod)

**Comment by GeekyGami** [comment]
> he is remaking the whole thing from scratch, that probably takes a lot of work to make. i doubt he's working on it 24/7, he probably has other projects/work/school. Also keep in mind he isn't paid a penny for doing this as far as i know, so il looks more like a hobby project than anything else. (he also could of just lost interest in dolphinVR, but that sounds rather unlikely looking at the other stuff he made that looked tedious AF to do, like the doom 3 VR mod)

... Okay, sure, but isn't it starting to get a tiny bit odd?
It didn't take this long initially when he made the original versions, right?
Why hasn't anything been said yet, it's been even more months since.

Even Nibre's VR project for MCC we got news from later about him getting employed by Microsoft

**Comment by thatsprettygood** [comment]
He hasn't spoken a word since October. Not even an comment.

**Comment by isamu99** [comment]
Maybe he's dead

**Comment by thatsprettygood** [comment]
Yep
![image](https://user-images.githubusercontent.com/36249601/133680511-52ca0ed4-de04-4bc5-b14f-7ea104289ffb.png)

**Comment by GeekyGami** [comment]
> Yep ![image](https://user-images.githubusercontent.com/36249601/133680511-52ca0ed4-de04-4bc5-b14f-7ea104289ffb.png)

Small note, as far as I'm aware, he's using a different account and a different fork for the changes he's working on.
The account is basically only there to get in touch here, but is otherwise dormant.

The reason we can't tell what's going on is because of that.

Ah, we're past 1 year since his last message.

**Comment by isamu99** [comment]
Have you been able to deduce any information on whether he's actually alive or not?

**Comment by thatsprettygood** [comment]
> Small note, as far as I'm aware, he's using a different account and a different fork for the changes he's working on. The account is basically only there to get in touch here, but is otherwise dormant.
> 
> The reason we can't tell what's going on is because of that.
> 
> Ah, we're past 1 year since his last message.

How do you know he is working on another account?

**Comment by GeekyGami** [comment]
> 

Well, first of all, he said so himself.
Second of all, he's banned from making contributions to the main Dolphin repo, as this branch was going against the GPLV3 license they were using for one reason or another, and because of political turmoil.
Due to that he's been making his changes on a separate branch he hasn't shared.

This is also the case because he hasn't contributed anything on this account since prior to announcing that,
and because he has said very clearly that he had to start re-implementation from scratch due to there being too many differences in the newer versions of Dolphin.

**Comment by thatsprettygood** [comment]
Any.. updates?

Edit: He's alive!
